src/coreclr/src/jit/assertionprop.cpp

```diff
@@ -2581,21 +2578,33 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
     }
 }
 
-/*******************************************************************************************************
- *
- * Perform constant propagation on a tree given the "curAssertion" is true at the point of the "tree."
- *
- */
-GenTree* Compiler::optConstantAssertionProp(AssertionDsc* curAssertion,
-                                            GenTree*      tree,
+//------------------------------------------------------------------------------
+// optConstantAssertionProp: Possibly substitute a constant for a local use
+//
+// Arguments:
+//    curAssertion - assertion to propagate
+//    tree         - tree to possibly modify
+//    stmt         - statement containing the tree
+//    index        - index of this assertion in the assertion table
+//
+// Returns:
+//    Updated tree (may be the input tree, modified in place), or nullptr
+//
+// Notes:
+//    stmt may be nullptr during local assertion prop
+//
+GenTree* Compiler::optConstantAssertionProp(AssertionDsc*        curAssertion,
+                                            GenTreeLclVarCommon* tree,
                                             Statement* stmt DEBUGARG(AssertionIndex index))
 {
-    unsigned lclNum = tree->AsLclVarCommon()->GetLclNum();
+    const unsigned lclNum = tree->GetLclNum();
 
+#if FEATURE_ANYCSE
```

Comment:
We will probably want to delete all occurrences of FEATURE_ANYCSE as some point in the future.